lib/events.js

```diff
@@ -163,24 +161,21 @@ function emitMany(handler, isFn, self, args) {
   }
 }
 
-EventEmitter.prototype.emit = function emit(type) {
-  var er, handler, len, args, i, events, domain;
-  var needDomainExit = false;
-  var doError = (type === 'error');
+EventEmitter.prototype.emit = function emit(type, ...args) {
```

Comment:
@bmeurer what do you think? As far as I understand, this is related to the escape analysis system. Do we need V8 62 to have the perf benefit?

Reply:
The escape analysis is responsible for avoiding allocations of either `arguments` object or rest parameters. If escape analysis is off, every invocation of this function will always have to allocate `arguments` or rest parameters, even if it's *not used*.

So, yes, for code like this, independent of whether you use `arguments` or rest parameters, escape analysis usually helps performance-wise.

Reply:
What is the latest on Escape Analysis in 8.x LTS? I know there was an issue opened discussing it but can't tell by looking at it what was decided, if anything yet.